app/tests/cases_tests/test_widget.py

```diff
@@ -161,3 +173,102 @@ def test_flexible_image_widget_prepopulated_value():
     assert field.initial is None
+
+
+@pytest.mark.django_db
+def test_dicom_upload_field_validation():
+    user = UserFactory()
+    ci = ComponentInterfaceFactory()
+    field = DICOMUploadField(user=user)
+    upload1 = UserUploadFactory(creator=user)
+    upload1.status = UserUpload.StatusChoices.COMPLETED
+    upload1.save()
+    upload2 = UserUploadFactory()
+    upload2.status = UserUpload.StatusChoices.COMPLETED
+    upload2.save()
+
+    dicom_upload = DICOMUploadWithName(
+        name="test_image",
+        user_uploads=[str(upload1.pk)],
+    )
+    parsed_value_for_upload_from_user = field.widget.value_from_datadict(
+        data={
+            f"{INTERFACE_FORM_FIELD_PREFIX}{ci.slug}_{DICOMUploadWidgetSuffixes[1]}": [
+                str(upload1.pk)
+            ],
+            f"{INTERFACE_FORM_FIELD_PREFIX}{ci.slug}_{DICOMUploadWidgetSuffixes[0]}": "test_image",
+        },
+        name=f"{INTERFACE_FORM_FIELD_PREFIX}{ci.slug}",
+        files={},
+    )
+    decompressed_value_for_upload_from_user = field.widget.decompress(
+        dicom_upload
+    )
+    assert (
+        parsed_value_for_upload_from_user
+        == decompressed_value_for_upload_from_user
+        == ["test_image", [str(upload1.pk)]]
+    )
+    assert field.clean(parsed_value_for_upload_from_user) == dicom_upload
+
+    dicom_upload_2 = DICOMUploadWithName(
+        name="test_image_2",
+        user_uploads=[str(upload2.pk)],
+    )
+    parsed_value_from_upload_from_other_user = field.widget.value_from_datadict(
+        data={
+            f"{INTERFACE_FORM_FIELD_PREFIX}{ci.slug}_{DICOMUploadWidgetSuffixes[1]}": [
+                str(upload2.pk)
+            ],
+            f"{INTERFACE_FORM_FIELD_PREFIX}{ci.slug}_{DICOMUploadWidgetSuffixes[0]}": "test_image_2",
+        },
+        name=f"{INTERFACE_FORM_FIELD_PREFIX}{ci.slug}",
+        files={},
+    )
+    decompressed_value_for_upload_from_other_user = field.widget.decompress(
+        dicom_upload_2
+    )
+    assert (
+        parsed_value_from_upload_from_other_user
+        == decompressed_value_for_upload_from_other_user
+        == ["test_image_2", [str(upload2.pk)]]
+    )
+    with pytest.raises(ValidationError):
+        field.clean(parsed_value_from_upload_from_other_user)
+
+
+@pytest.mark.django_db
+def test_dicom_upload_widget_prepopulated_value(mocker):
+    user_with_perm, user_without_perm = UserFactory.create_batch(2)
+    upload = DICOMImageSetUploadFactory()
+    upload.user_uploads.set([UserUploadFactory(creator=user_with_perm)])
+    im = ImageFactory(
+        name="test_image",
+        dicom_image_set=DICOMImageSetFactory(dicom_image_set_upload=upload),
+    )
+    assign_perm("cases.view_image", user_with_perm, im)
+    ci = ComponentInterfaceFactory(kind=ComponentInterface.Kind.PANIMG_IMAGE)
+    civ = ComponentInterfaceValueFactory(interface=ci, image=im)
+
+    # fake DICOM ci for now
+    # TODO: create a proper DICOM super kind ci above

```

Comment:
This needs to be resolved.

Reply:
This can only be resolved after this cycle since it depends on the new kind and all surrounding logic to be in place.

Reply:
As far as I can tell it only depends on the key being present and set. Why would it depend on the surrounding logic?

Reply:
Isn't the only thing missing for that in #4318:

```python3
    @property
    def is_dicom_image_kind(self):
        return self.kind == InterfaceKindChoices.DICOM_IMAGE_SET
```

Reply:
For this test, yes, that would be enough.